clippy_lints/src/methods/unwrap_or_else_default.rs

```diff
@@ -44,3 +52,19 @@ pub(super) fn check<'tcx>(
         }
     }
 }
+
+fn closure_body_returns_empty_to_string(body: &hir::Body<'_>) -> bool {
```

Comment:
Can this method receive `cx` and `u_arg`, and these lines https://github.com/rust-lang/rust-clippy/pull/9421/files#diff-eee44b9635bbab1626371d693f19dd5366810f29fa80e83f6e053c368ef09785R27-R32 be put into this? For example, as follows:
```rust
fn closure_body_returns_empty_to_string(cx: &LateContext<'_>, e: &hir::Expr<'_>) -> bool {
    if let hir::ExprKind::Closure(&hir::Closure { body, .. }) = e.kind {
        let body = cx.tcx.hir().body(body);

        if body.params.is_empty()
            && let hir::Expr{ kind, .. } = &body.value
            && let hir::ExprKind::MethodCall(hir::PathSegment {ident, ..}, [self_arg], _) = kind
            && ident == &symbol::Ident::from_str("to_string")
            && let hir::Expr{ kind, .. } = self_arg
            && let hir::ExprKind::Lit(lit) = kind
            && let LitKind::Str(symbol::kw::Empty, _) = lit.node
        {
            return true;
        }
    }

    false
}
```

Reply:
https://github.com/rust-lang/rust-clippy/compare/0160e12edd98423a3b1b69e07688930d57ac5aab..b21d9d307b5fdb1e423d786f9899757b42957b07#diff-eee44b9635bbab1626371d693f19dd5366810f29fa80e83f6e053c368ef09785R49-R62